MaxText/layers/attentions.py

```diff
@@ -260,22 +241,174 @@ def key_rotary(self, key: Array, inputs_positions: Array):
       name='key_rotary')(inputs=key, position=inputs_positions)
     return key
 
-  def apply_attention(
+
+  # Following Pallas MHA Flash Attention Reference.
+  # https://github.com/google/jax/blob/main/jax/experimental/pallas/ops/tpu/flash_attention.py
+  def generate_attention_mask(
```

Comment:
No docstring.

Reply:
Legitimate feedback but I'm just trying to move things around here! Thanks for calling it out.